data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java

```diff
@@ -144,17 +145,23 @@ public BeaconRestApi(final DataProvider dataProvider, final TekuConfiguration co
               config.logIfServerNotStarted = false;
               config.showJavalinBanner = false;
             });
+    restApiEnabled = configuration.isRestApiEnabled();
     initialize(dataProvider, configuration);
   }
 
   BeaconRestApi(
       final DataProvider dataProvider, final TekuConfiguration configuration, final Javalin app) {
     this.app = app;
+    this.restApiEnabled = configuration.isRestApiEnabled();
     initialize(dataProvider, configuration);
   }
 
   public void start() {
-    app.start();
+    if (restApiEnabled) {
+      app.start();
```

Comment:
Can we avoid creating `BeaconRestApi` entirely if it's disabled?

Reply:
i'll see how ugly it gets, I was trying to minimise the change but maybe.